nixos/modules/system/boot/loader/grub/grub.nix

```
@@ -738,13 +746,23 @@ in
             XMLLibXML XMLSAX XMLSAXBase
             ListCompare JSON
           ]);
-        in pkgs.writeScript "install-grub.sh" (''
-        #!${pkgs.runtimeShell}
-        set -e
-        ${optionalString cfg.enableCryptodisk "export GRUB_ENABLE_CRYPTODISK=y"}
-      '' + flip concatMapStrings cfg.mirroredBoots (args: ''
-        ${perl}/bin/perl ${install-grub-pl} ${grubConfig args} $@
-      '') + cfg.extraInstallCommands);
+          ng = pkgs.install-grub-ng.override {
+            inherit (config.system.nixos) distroName;
+          };
+          genRun = args: if cfg.useInstallNg then
+            "${lib.getExe ng} ${grubConfig args} $@\n"
+          else
+            "${lib.getExe perl} ${install-grub-pl} ${grubConfig args} $@\n";
```

Might as well fix the shell quoting issue:
```suggestion
            ''${lib.getExe ng} ${grubConfig args} "$@"\n''
          else
            ''${lib.getExe perl} ${install-grub-pl} ${grubConfig args} "$@"\n'';
```